x-pack/plugins/apm/common/runtime_types/agent_configuration_intake_rt/index.ts

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import * as t from 'io-ts';
+import { transactionSampleRateRt } from '../transaction_sample_rate_rt';
+import { transactionMaxSpansRt } from '../transaction_max_spans_rt';
+
+export const serviceRt = t.partial({
+  name: t.string,
+  environment: t.string
+});
+
+export const agentConfigurationIntakeRt = t.intersection([
+  t.partial({ agent_name: t.string }),
```

Comment:
for consistency, should we use `{ agent: { name } }` here instead of `{ agent_name }`?

Reply:
I'd prefer `agent.name` but I think that ship has already sailed since it's stored as that in ES. We could change the API only but isn't it odd to have it as `{ agent: { name } }` on the api and then `{ agent_name }` in es?

Reply:
yeah, I can see that! not worth it then.